src/js/chartAuthoring.js

```diff
@@ -188,6 +245,72 @@ https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.tx
         }]
     });
 
+    floe.chartAuthoring.updateTitleFromEdit = function(pieChart, newTitle) {
+        pieChart.applier.change("pieTitle", newTitle);
+    };
+
+    floe.chartAuthoring.updateDescriptionFromEdit = function(pieChart, newDescription) {
+        pieChart.applier.change("pieDescription", newDescription);
+    };
+
+    // Adds and removes highlight
+    floe.chartAuthoring.highlightPlayingData = function(that) {
+        var currentlyPlayingData = that.model.currentlyPlayingData;
+        var chartAuthoringInterface = that.chartAuthoringInterface;
+
+        // The chart authoring interface is ready
+        if(chartAuthoringInterface !== undefined) {
+            var legendTable, tbody, rows, pie, slices;
+            var dataPlayingHighlightClass = that.options.strings.dataPlayingHighlightClass;
+            legendTable = chartAuthoringInterface.pieChart.legend.table;
+
+            pie = chartAuthoringInterface.pieChart.pie;
+            slices = pie.paths;
+
+            tbody = legendTable.selectAll("tbody");
+            rows = tbody.selectAll("tr");
+
+            // Nothing is currently playing; remove any highlighting
+            if(currentlyPlayingData === null) {
+                rows.classed(dataPlayingHighlightClass,false);
+                slices.classed(dataPlayingHighlightClass,false);
+            }
+
+            // Highlight pie chart slices / legend rows as they play
+            if(currentlyPlayingData !== null) {
+
+                var activeRow = rows.filter(
```

Comment:
The action taken for rows and slices are pretty much same, can we create a function to share?